src/windows/functions.cpp

```diff
@@ -233,12 +233,26 @@ BOOL EjectRemovableVolume(HANDLE volume) {
     return FALSE;
   }
 
-  return DeviceIoControl(volume,
-                         IOCTL_STORAGE_EJECT_MEDIA,
-                         NULL, 0,
-                         NULL, 0,
-                         &bytesReturned,
-                         NULL);
+  size_t retries = 5;
+
+  while (retries > 0) {
+    const BOOL result = DeviceIoControl(volume,
+                                        IOCTL_STORAGE_EJECT_MEDIA,
+                                        NULL, 0,
+                                        NULL, 0,
+                                        &bytesReturned,
+                                        NULL);
+    if (result) {
```

Comment:
Minor nit, but it looks like this could have just been
```
    if (DeviceIoControl(volume,
                        IOCTL_STORAGE_EJECT_MEDIA,
                        NULL, 0,
                        NULL, 0,
                        &bytesReturned,
                        NULL)) {
```
?
(although perhaps such a multi-line `if` is a bit less readable :wink: )

Reply:
Yeah, I moved it out since I felt it was not very readable.